src/components/Modal/__stories__/helpers.js

```diff
@@ -0,0 +1,39 @@
+import { Button, Flex } from "components";
+import React, { useRef, useState } from "react";
+
+// internal custom hook to help with writing tests and stories.
+export const useModalHelper = ({ openButtonTitle = "Open", openOnStart = false, ...props } = {}) => {
```

Comment:
The problem with code inside helpers in stories is that the people who read the story can't see its implementation (and therefore they usually not understand what is this helper and where they can find it).
In my opinion even this code is reusable, the readability is more important in these cases (stories).  So it will be better to copy past this code to each story (and it will be helpful for developers which trying to understand who to use the modal)